beetsplug/yamleditor.py

```diff
@@ -0,0 +1,329 @@
+# This file is part of beets.
+# Copyright 2015, Jean-Marie Winters
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""open tags of items in texteditor,change them and save back to the items.
+"""
+from __future__ import (division, absolute_import, print_function,
+                        unicode_literals)
+
+from beets import plugins
+from beets.ui import Subcommand, decargs, library, print_
+import subprocess
+import difflib
+import yaml
+import collections
+import webbrowser
+from sys import exit
+from beets import config
+from beets import ui
+from tempfile import NamedTemporaryFile
+
+
+class yamleditorPlugin(plugins.BeetsPlugin):
+
+    def __init__(self):
+        super(yamleditorPlugin, self).__init__()
+
+        self.config.add({
+            'style': 'yaml',
+            'editor': '',
+            'diff_method': '',
+            'html_viewer': '',
+            'editor_args': '',
+            'html_args': '',
+            'albumfields': 'album albumartist',
+            'itemfields': 'track title artist album ',
+            'not_fields': 'path',
+            'separator': '-'
+
+        })
+
+    def commands(self):
+        yamleditor_command = Subcommand(
+            'yamleditor',
+            help='send items to yamleditor for editing tags')
+        yamleditor_command.parser.add_option(
+            '-e', '--extra',
+            action='store',
+            help='add additional fields to edit',
+        )
+        yamleditor_command.parser.add_all_common_options()
+        yamleditor_command.func = self.editor_music
+        return[yamleditor_command]
+
+    def editor_music(self, lib, opts, args):
+        """edit tags in a textfile in yaml-style
+        """
+        self.style = self.config['style'].get()
+        """the editor field in the config lets you specify your editor.
```

Comment:
When they appear in the middle of a function, these are best written as comments `# Foo.` rather than docstrings `"""Foo."""`.